backend/app/controllers/profileController.ts

```diff
@@ -85,6 +72,54 @@ class ProfileController {
     return res.status(200).json(payload)
   }
 
+  public getFollowing = async (req: Request, res: Response): Promise<Response> => {
+    const receivedUsername = this.getUsernameFromToken(req)
+
+    if (!receivedUsername) {
+      res.status(404).json({ error: 'No username was given' })
+    }
+
+    const profile = await this.dao.getProfileByUsername(receivedUsername)
+    const followingById = await this.dao.getFollowingByProfileId(profile.id)
+
+    const payload = {
+      following: []
+    }
+
+    followingById.forEach((entry) => {
+      payload.following.push(entry.profile)
+    })
+
+    if (payload.following.length < 1) {
+      return res.status(204)
+    }
+
+    return res.status(200).json(payload)
+  }
+
+  private getUsernameFromToken = (req: Request) => {
+    const { token } = req.cookies
+    let decodedToken: any
+
+    try {
+      decodedToken = jwt.verify(token, process.env.JWT_SECRET)
+    } catch (e) {
+      decodedToken = null
+    }
+
+    let receivedUsername = req.params.username
+
+    if (!receivedUsername) {
+      if (decodedToken && decodedToken.username) {
+        receivedUsername = decodedToken.username
+      } else {
+        return null
+      }
+    }
+
+    return receivedUsername
```

Comment:
```suggestion
    if (!receivedUsername && decodedToken?.username) {
      receivedUsername = decodedToken.username
    }

    return receivedUsername || null
```

Reply:
Actually no wait. This can be made even simpler.

Reply:
```suggestion
    return receivedUsername || decodedToken?.username
```

If `receivedUsername` is falsy (Might want to decide whether to use ?? or || here), you go to `decodedToken?.username`. If decodedToken is null, this will also result in null. Meaning that you return null as you intended in your code.